offscreen-canvas/tools/tests2d.yaml

```diff
@@ -9663,8 +10139,8 @@
     - CanvasTest
   code: |
     ctx.font = '50px CanvasTest';
-    deferTest();
-    step_timeout(t.step_func_done(function () {
+    new Promise(function(resolve) { step_timeout(resolve, 500); })
+      .then(function() {
```

Comment:
Shouldn't promise functions like this use t.step_func so the assertions execute within the test harness wrapper?

Reply:
We could wrap the handlers, but as long as we're willing to trust the Promise implementation, we'll detect and report the failure with the final invocation on the Promise chain: `.then(t_pass, t_fail)`

Reply:
Ah, that's a good point!